src/library_manager/lib_manager.c

```diff
@@ -69,6 +69,68 @@ static int lib_manager_load_data_from_storage(void __sparse_cache *vma, void *s_
 	return 0;
 }
 
+static int lib_manager_load_module(uint32_t module_id, struct sof_man_module *mod,
+				   struct sof_man_fw_desc *desc);
+static int lib_manager_unload_module(uint32_t module_id, struct sof_man_module *mod,
+				     struct sof_man_fw_desc *desc);
+
+static int lib_manager_load_aux(struct sof_man_module *mod,
+				struct sof_man_fw_desc *desc, uint32_t lib_id)
+{
+	struct ext_library *ext_lib = ext_lib_get();
+
+	/* There are modules marked as lib_code. This is code shared between several modules inside
+	 * the library. Load all lib_code modules with first none lib_code module load.
+	 */
+	if (!mod->type.lib_code)
+		ext_lib->mods_exec_load_cnt++;
+
+	if (ext_lib->mods_exec_load_cnt == 1) {
+		struct sof_man_module *module_entry =
+		    (struct sof_man_module *)((char *)desc + SOF_MAN_MODULE_OFFSET(0));
+
+		for (size_t idx = 0; idx < desc->header.num_module_entries;
```

Comment:
why the size_t for an index incremented by one?

Reply:
@plbossart no idea, not my code, moved it as is from the current implementation